### DISCUSSION TITLE
Multiple pages not working react

### DISCUSSION THREAD

**Discussion by zamachnoi** [community/community#35639] [Other Feature Feedback, Questions, & Ideas] [answered]
I have multiple pages for my gh-pages site, but only one works on load.

my site: https://nickolasz523.github.io/

my repo: https://github.com/nickolasz523/nickolasz523.github.io

edit: it does work with netlify tho... https://stoic-yonath-6fbdb6.netlify.app/home

**Comment by jfullstackdev** [comment]
seems to be working fine here

![image](https://user-images.githubusercontent.com/47092464/194878408-c6e28cab-6750-46e9-b10f-1334235bf83c.png)

there are no error logs

**Reply by zamachnoi** [reply]
But when I click home or about or contact it doesn't work sadly

**Comment by igor-h** [comment, marked as answer]
Hi @nickolasz523 

If deploying to GitHub, ensure there is a "homepage" entry for where you are hosting it in Github.

Example:
` "homepage": "https://nickolasz523.github.io/home",`

Switch to the HashRouter since GitHub pages doesn't support the tech used by the BrowserRouter.

`index.js`
```
import React from 'react';
 import ReactDOM from 'react-dom/client';
 import { HashRouter } from 'react-router-dom'
 import App from './App';
 import './styles/index.css';

 ReactDOM.createRoot(document.getElementById('root')).render(
   <React.StrictMode>
     <HashRouter>
       <App />
     </HashRouter>
   </React.StrictMode>
 );
```

For more details see the `create-react-app` docs for https://create-react-app.dev/docs/deployment/#github-pages
and
https://create-react-app.dev/docs/deployment/#notes-on-client-side-routing

Medium article about this topic:
https://dev.to/janjibdev/problem-with-react-router-app-and-github-pages-lij


_If this was helpful mark my comment as Resolved._
Thanks, Igor. 👍🏽

**Reply by igor-h** [reply]
Hey @nickolasz523 was this useful for you?
